docs/src/extend/custom-rules.md

```diff
@@ -614,37 +614,161 @@ You can also access comments through many of `sourceCode`'s methods using the `i
 
 Rules may export a `schema` property, which is a [JSON Schema](https://json-schema.org/) format description of a rule's options which will be used by ESLint to validate configuration options and prevent invalid or unexpected inputs before they are passed to the rule in `context.options`.
 
-There are two formats for a rule's exported `schema`:
+If `schema` is not specified, then any options (besides severity) supplied to the rule are passed on verbatim without validation. This is a legacy behaviour though, and all rules are encouraged to specify a schema to aid developers when configuring your rules. Note that in v9, this behaviour will be deprecated entirely, and not specifying `schema` will force the rule to have no options (equivalent to `[]`) - see the [Require schemas and object-style rules](https://github.com/eslint/rfcs/blob/main/designs/2021-schema-object-rules/README.md) RFC.
+
+When validating a rule's options, there are 3 main preparation/validation steps:
 
-1. A full JSON Schema object describing all possible options the rule accepts.
-2. An array of JSON Schema objects for each optional positional argument.
+1. If the options are not an array, the value is wrapped into an array (e.g. `"off"` becomes `["off"]`)
+2. ESLint will validate the first element of the options array as a severity (`"off"`, `"warn"`, `"error"`, `0`, `1`, `2`)
+3. If the rule is enabled, then the remaining elements of the array are sliced and the rule's schema validation is run on this sliced options array (e.g. `["warn", "never", { someOption: 5 }]` results in the schema being applied to `["never", { someOption: 5 }]`)
 
-In both cases, these should exclude the [severity](../use/configure/rules#rule-severities), as ESLint automatically validates this first.
+Note that this means that the rule schema cannot see or validate the severity, it only sees the array elements _after_ the severity.
+
+There are two formats for a rule's exported `schema`:
+
+* An array of JSON Schema objects
+    * Each element will be checked against the same position in the sliced options array
+    * If the sliced options array has fewer elements than there are schemas, then the unmatched schemas are ignored
+    * If the sliced options array has more elements than there are schemas, the validation fails
+    * Two important consequences:
+        * Empty rule options beyond severity are always valid (e.g. `["error"]` will be a valid rule option)
+        * An empty array _enforces no options_ for the rule
```

Comment:
I've re-read this a few times and I'm still not clear on what these lines are trying to say. 

Consequences of what?

Is the first bullet saying that you are not required to pass options beyond severity?

For the second bullet, what is enforcing no options? An empty array where?

Reply:
Sure, I've tried to clarify the wording on this to highlight the always valid/always error cases a bit more:

>    * There are two important consequences to using this format:
>        * It is _always valid_ for a user to provide no options to your rule (beyond severity)
>        * If you specify an empty array, then it is _always an error_ for a user to provide any options to your rule (beyond severity)